atom/browser/atom_browser_main_parts_mac.mm

```diff
@@ -6,13 +6,53 @@
 
 #include "atom/browser/atom_paths.h"
 #include "atom/browser/mac/atom_application_delegate.h"
-#include "base/mac/bundle_locations.h"
-#include "base/mac/foundation_util.h"
-#include "base/path_service.h"
-#include "ui/base/l10n/l10n_util_mac.h"
 
 namespace atom {
 
+namespace {
+
+base::scoped_nsobject<NSMenuItem> CreateMenuItem(NSString* title,
+                                                 SEL action,
+                                                 NSString* key_equivalent) {
+  return base::scoped_nsobject<NSMenuItem>([[NSMenuItem alloc]
+      initWithTitle:title
+             action:action
+      keyEquivalent:key_equivalent]);
+}
+
+// The App Menu refers to the dropdown titled "Electron".
+base::scoped_nsobject<NSMenu> BuildAppMenu() {
+  // The title is not used, as the title will always be the name of the App.
+  base::scoped_nsobject<NSMenu> menu([[NSMenu alloc] initWithTitle:@""]);
+
+  NSString* app_name = [[[NSBundle mainBundle] infoDictionary]
+      objectForKey:(id)kCFBundleNameKey];
+
+  base::scoped_nsobject<NSMenuItem> item =
+      CreateMenuItem([NSString stringWithFormat:@"Quit %@", app_name],
+                     @selector(terminate:), @"q");
+  [menu addItem:item];
+
+  return menu;
+}
+
+base::scoped_nsobject<NSMenu> BuildEmptyMainMenu() {
+  base::scoped_nsobject<NSMenu> main_menu([[NSMenu alloc] initWithTitle:@""]);
+
+  using Builder = base::scoped_nsobject<NSMenu> (*)();
+  static const Builder kBuilderFuncs[] = {&BuildAppMenu};
```

Comment:
Is this copied from somewhere? it seems like overkill

Reply:
Yeah @nornagon I deliberately copied the pattern from Chromium so that future changes to anything like this are easily comparable.